### ISSUE TITLE
Test fails

### ISSUE THREAD

**Issue by sander85** [plicease-legacy/Dist-Zilla-PluginBundle-Author-Plicease#7]
Hi!

When trying to build the latest version I'm getting this result:

```
t/00_diag.t ... ok
t/01_use.t .... ok
        # directory [Foo-Bar] does not exist

        #   Failed test 'dir Foo-Bar exists'
        #   at /usr/lib/perl5/vendor_perl/5.22.0/Test/Dir/Base.pm line 54.
        # File [Foo-Bar/dist.ini] does not exist

        #   Failed test 'Foo-Bar/dist.ini exists'
        #   at t/mint.t line 49.
        # File [Foo-Bar/lib/Foo/Bar.pm] does not exist

        #   Failed test 'Foo-Bar/lib/Foo/Bar.pm exists'
        #   at t/mint.t line 50.
        # Looks like you failed 3 tests of 3.

    #   Failed test 'new'
    #   at t/mint.t line 51.
    # Looks like you failed 1 test of 3.

#   Failed test 'dzil'
#   at t/mint.t line 111.
# Looks like you failed 1 test of 1.
t/mint.t ...... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 
t/thanks1.t ... ok
t/thanks2.t ... ok
t/thanks3.t ... ok
t/thanks4.t ... ok
t/unbundle.t .. ok

Test Summary Report
-------------------
t/mint.t    (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
Files=8, Tests=28,  7 wallclock secs ( 0.04 usr  0.01 sys +  6.06 cusr  0.23 csys =  6.34 CPU)
Result: FAIL
Failed 1/8 test programs. 1/28 subtests failed.
```

**Comment by plicease** [comment]
The output of 00_diag.t would be helpful to reproduce this.

**Comment by sander85** [comment]
Here you go:

```
# HARNESS_ACTIVE=1
# HARNESS_VERSION=3.35
# LANG=C
# LC_SOURCED=1
# PERL5LIB=/home/sander/rpmbuild/BUILD/Dist-Zilla-PluginBundle-Author-Plicease-1.81/blib/lib:/home/sander/rpmbuild/BUILD/Dist-Zilla-PluginBundle-Author-Plicease-1.81/blib/arch:inc
# PERL_DL_NONLAZY=1
# SHELL=/bin/bash
# 
# 
# 
# PERL5LIB path
# /home/sander/rpmbuild/BUILD/Dist-Zilla-PluginBundle-Author-Plicease-1.81/blib/lib
# /home/sander/rpmbuild/BUILD/Dist-Zilla-PluginBundle-Author-Plicease-1.81/blib/arch
# inc
# 
# 
# 
# perl                                    v5.22.0
# Capture::Tiny                           0.25
# Dist::Zilla                             5.037
# Dist::Zilla::App                        5.037
# Dist::Zilla::Plugin::AutoMetaResources  1.21
# Dist::Zilla::Plugin::CopyFilesFromBuild 0.151680
# Dist::Zilla::Plugin::InsertExample      0.04
# Dist::Zilla::Plugin::InstallGuide       1.200006
# Dist::Zilla::Plugin::MinimumPerl        1.006
# Dist::Zilla::Plugin::OurPkgVersion      0.06
# Dist::Zilla::Plugin::PodWeaver          4.006
# Dist::Zilla::Plugin::ReadmeAnyFromPod   0.150250
# Dist::Zilla::Plugin::Run::BeforeBuild   0.039
# File::chdir                             0.1010
# IPC::System::Simple                     1.25
# JSON::PP                                2.27300
# Module::Build                           0.4214
# Moose                                   2.1405
# Path::Class                             0.35
# PerlX::Maybe                            1.001
# PerlX::Maybe::XS                        1.001
# Pod::Markdown                           2.002
# Test::Dir                               1.014
# Test::File                              1.44
# Test::File::ShareDir                    1.001001
# Test::Fixme                             0.14
# Test::More                              1.001014
# Test::Pod                               1.51
# Test::Pod::Coverage                     1.10
# Test::Script                            1.10
# Test::Version                           2.03
# YAML                                    1.15
# YAML::XS                                0.59
# autodie                                 2.26
```

**Comment by plicease** [comment]
Okay it doesn't appear to be a dep issue, can you give me the verbose output of mint.t:

prove -lv t/mint.t

Thanks!

**Comment by sander85** [comment]
Thanks for your help!

TL;DR version: the following modules were also needed:
Dist::Zilla::Plugin::Git::Check
Perl::PrereqScanner
Term::Encoding

The long version:

```
$ prove -lv t/mint.t 
t/mint.t .. 
1..1
    # Subtest: dzil
    1..3
        # Subtest: new
        1..3
        # + new -P Author::Plicease Foo::Bar
        # EXCEPTION:Can't locate Term/Encoding.pm in @INC (you may need to install the Term::Encoding module) (@INC contains: /tmp/uaxp6dIJcy /home/sander/rpmbuild/BUILD/Dist-Zilla-PluginBundle-Author-Plicease-1.81/lib /usr/lib/perl5/site_perl/5.22.0/x86_64-linux-thread-multi /usr/lib/perl5/site_perl/5.22.0 /usr/lib/perl5/vendor_perl/5.22.0/x86_64-linux-thread-multi /usr/lib/perl5/vendor_perl/5.22.0 /usr/lib/perl5/5.22.0/x86_64-linux-thread-multi /usr/lib/perl5/5.22.0 /usr/lib/perl5/site_perl /usr/lib/perl5/vendor_perl/5.22.0 /usr/lib/perl5/vendor_perl/5.22.0/x86_64-linux-thread-multi /usr/lib/perl5/vendor_perl/5.20.1 /usr/lib/perl5/vendor_perl /home/sander/rpmbuild/BUILD/Dist-Zilla-PluginBundle-Author-Plicease-1.81) at /usr/lib/perl5/vendor_perl/5.22.0/Dist/Zilla/Chrome/Term.pm line 58.
        # 
        not ok 1 - dir Foo-Bar exists
        not ok 2 - Foo-Bar/dist.ini exists
        not ok 3 - Foo-Bar/lib/Foo/Bar.pm exists
        # directory [Foo-Bar] does not exist

        #   Failed test 'dir Foo-Bar exists'
        #   at /usr/lib/perl5/vendor_perl/5.22.0/Test/Dir/Base.pm line 54.
        # File [Foo-Bar/dist.ini] does not exist

        #   Failed test 'Foo-Bar/dist.ini exists'
        #   at t/mint.t line 49.
        # File [Foo-Bar/lib/Foo/Bar.pm] does not exist

        #   Failed test 'Foo-Bar/lib/Foo/Bar.pm exists'
        #   at t/mint.t line 50.
        # Looks like you failed 3 tests of 3.
    not ok 1 - new

    #   Failed test 'new'
    #   at t/mint.t line 51.
        # Subtest: build
        1..0 # SKIP previous step failed
    ok 2 # skip previous step failed
        # Subtest: test
        1..0 # SKIP previous step failed
    ok 3 # skip previous step failed
    # Looks like you failed 1 test of 3.
not ok 1 - dzil

#   Failed test 'dzil'
#   at t/mint.t line 111.
# Looks like you failed 1 test of 1.
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 

Test Summary Report
-------------------
t/mint.t (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
Files=1, Tests=1,  1 wallclock secs ( 0.02 usr  0.01 sys +  0.65 cusr  0.02 csys =  0.70 CPU)
Result: FAIL
```

I installed Term::Encoding

```
$ prove -lv t/mint.t 
t/mint.t .. 
1..1
    # Subtest: dzil
    1..3
        # Subtest: new
        1..3
        # + new -P Author::Plicease Foo::Bar
        # [DZ] making target dir /tmp/70M6hSw9jz/Foo-Bar
        # [DZ] writing files to /tmp/70M6hSw9jz/Foo-Bar
        # [DZ] no Git::Wrapper, can't create repository
        # [DZ] dist minted in ./Foo-Bar
        ok 1 - dir Foo-Bar exists
        ok 2 - Foo-Bar/dist.ini exists
        ok 3 - Foo-Bar/lib/Foo/Bar.pm exists
    ok 1 - new
        # Subtest: build
        1..1
        # + build
        # EXCEPTION:Required plugin Dist::Zilla::Plugin::Git::Check isn't installed.
        # 
        # Run 'dzil authordeps' to see a list of all required plugins.
        # You can pipe the list to your CPAN client to install or update them:
        # 
        #     dzil authordeps --missing | cpanm
        # 
        # 
        not ok 1 - Foo-Bar-0.01 exists
    not ok 2 - build
        # Subtest: test
        1..2
        # File [Foo-Bar-0.01] does not exist

        #   Failed test 'Foo-Bar-0.01 exists'
        #   at t/mint.t line 76.
        # Looks like you failed 1 test of 1.

    #   Failed test 'build'
    #   at t/mint.t line 80.
        # + test
        # EXCEPTION:Required plugin Dist::Zilla::Plugin::Git::Check isn't installed.
        # 
        # Run 'dzil authordeps' to see a list of all required plugins.
        # You can pipe the list to your CPAN client to install or update them:
        # 
        #     dzil authordeps --missing | cpanm
        # 
        # 
        not ok 1 - dir .build exists
        not ok 2 - dir .build is empty
    not ok 3 - test
not ok 1 - dzil
        # directory [.build] does not exist

        #   Failed test 'dir .build exists'
        #   at /usr/lib/perl5/vendor_perl/5.22.0/Test/Dir/Base.pm line 54.
        # directory [.build] is not empty

        #   Failed test 'dir .build is empty'
        #   at /usr/lib/perl5/vendor_perl/5.22.0/Test/Dir/Base.pm line 70.
        # Looks like you failed 2 tests of 2.

    #   Failed test 'test'
    #   at t/mint.t line 109.
    # Looks like you failed 2 tests of 3.

#   Failed test 'dzil'
#   at t/mint.t line 111.
# Looks like you failed 1 test of 1.
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 

Test Summary Report
-------------------
t/mint.t (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
Files=1, Tests=1,  1 wallclock secs ( 0.03 usr  0.00 sys +  1.25 cusr  0.04 csys =  1.32 CPU)
Result: FAIL
```

I installed Dist::Zilla::Plugin::Git::Check

```
$ prove -lv t/mint.t 
t/mint.t .. 
1..1
    # Subtest: dzil
    1..3
        # Subtest: new
        1..3
        # + new -P Author::Plicease Foo::Bar
        # [DZ] making target dir /tmp/XbOpsRihp9/Foo-Bar
        # [DZ] writing files to /tmp/XbOpsRihp9/Foo-Bar
        # EXCEPTION:
        # *** Please tell me who you are.
        # 
        # Run
        # 
        #   git config --global user.email "you@example.com"
        #   git config --global user.name "Your Name"
        # 
        # to set your account's default identity.
        # Omit --global to set the identity only in this repository.
        # 
        # fatal: empty ident name (for <sander@powerhouse.kodu>) not allowed
        # 
        ok 1 - dir Foo-Bar exists
        ok 2 - Foo-Bar/dist.ini exists
        ok 3 - Foo-Bar/lib/Foo/Bar.pm exists
    ok 1 - new
        # Subtest: build
        1..1
        # + build
        # [DZ] beginning to build Foo-Bar
        # [DZ] guessing dist's main_module is lib/Foo/Bar.pm
        # EXCEPTION:Can't locate Perl/PrereqScanner.pm in @INC (you may need to install the Perl::PrereqScanner module) (@INC contains: /tmp/UPuAMcaEFQ /home/sander/rpmbuild/BUILD/Dist-Zilla-PluginBundle-Author-Plicease-1.81/lib /usr/lib/perl5/site_perl/5.22.0/x86_64-linux-thread-multi /usr/lib/perl5/site_perl/5.22.0 /usr/lib/perl5/vendor_perl/5.22.0/x86_64-linux-thread-multi /usr/lib/perl5/vendor_perl/5.22.0 /usr/lib/perl5/5.22.0/x86_64-linux-thread-multi /usr/lib/perl5/5.22.0 /usr/lib/perl5/site_perl/5.22.0 /usr/lib/perl5/site_perl/5.22.0/x86_64-linux-thread-multi /usr/lib/perl5/site_perl /usr/lib/perl5/vendor_perl/5.22.0 /usr/lib/perl5/vendor_perl/5.22.0/x86_64-linux-thread-multi /usr/lib/perl5/vendor_perl/5.20.1 /usr/lib/perl5/vendor_perl /home/sander/rpmbuild/BUILD/Dist-Zilla-PluginBundle-Author-Plicease-1.81) at /usr/lib/perl5/vendor_perl/5.22.0/Dist/Zilla/Plugin/AutoPrereqs.pm line 124.
        # 
        not ok 1 - Foo-Bar-0.01 exists
    not ok 2 - build
        # Subtest: test
        1..2
        # File [Foo-Bar-0.01] does not exist

        #   Failed test 'Foo-Bar-0.01 exists'
        #   at t/mint.t line 76.
        # Looks like you failed 1 test of 1.

    #   Failed test 'build'
    #   at t/mint.t line 80.
        # + test
        # [DZ] building distribution under .build/mDQNUYtpOH for installation
        # [DZ] beginning to build Foo-Bar
        # [DZ] guessing dist's main_module is lib/Foo/Bar.pm
        # EXCEPTION:Can't locate Perl/PrereqScanner.pm in @INC (you may need to install the Perl::PrereqScanner module) (@INC contains: /tmp/UPuAMcaEFQ /home/sander/rpmbuild/BUILD/Dist-Zilla-PluginBundle-Author-Plicease-1.81/lib /usr/lib/perl5/site_perl/5.22.0/x86_64-linux-thread-multi /usr/lib/perl5/site_perl/5.22.0 /usr/lib/perl5/vendor_perl/5.22.0/x86_64-linux-thread-multi /usr/lib/perl5/vendor_perl/5.22.0 /usr/lib/perl5/5.22.0/x86_64-linux-thread-multi /usr/lib/perl5/5.22.0 /usr/lib/perl5/site_perl/5.22.0 /usr/lib/perl5/site_perl/5.22.0/x86_64-linux-thread-multi /usr/lib/perl5/site_perl /usr/lib/perl5/vendor_perl/5.22.0 /usr/lib/perl5/vendor_perl/5.22.0/x86_64-linux-thread-multi /usr/lib/perl5/vendor_perl/5.20.1 /usr/lib/perl5/vendor_perl /home/sander/rpmbuild/BUILD/Dist-Zilla-PluginBundle-Author-Plicease-1.81) at /usr/lib/perl5/vendor_perl/5.22.0/Dist/Zilla/Plugin/AutoPrereqs.pm line 124.
        # 
        ok 1 - dir .build exists
        not ok 2 - dir .build is empty
    not ok 3 - test
not ok 1 - dzil
        # directory [.build] is not empty

        #   Failed test 'dir .build is empty'
        #   at /usr/lib/perl5/vendor_perl/5.22.0/Test/Dir/Base.pm line 70.
        # Looks like you failed 1 test of 2.

    #   Failed test 'test'
    #   at t/mint.t line 109.
    # Looks like you failed 2 tests of 3.

#   Failed test 'dzil'
#   at t/mint.t line 111.
# Looks like you failed 1 test of 1.
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 

Test Summary Report
-------------------
t/mint.t (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
Files=1, Tests=1,  2 wallclock secs ( 0.03 usr  0.00 sys +  1.90 cusr  0.06 csys =  1.99 CPU)
Result: FAIL
```

I installed Perl::PrereqScanner and this time the build succeeded! \o/

**Comment by plicease** [comment]
\o/ \o/

Thanks.  I've released 1.82 which I think should skip the test when those modules aren't available.  I don't want to make [Git] a hard prereq, because it doesn't support 5.8.x or work well on MSWin32, but it is very useful for me to be able to do non-release type tasks in those environments.
